### ISSUE TITLE
Duplicate var declaration not reported with try..catch

### ISSUE THREAD

**Issue by romario333** [jshint/jshint#1399]
This happens with the latest jshint 2. To reproduce, paste this code to [jshint.com](http://jshint.com/):

```javascript
function test() {
  try {
    var result = 'a';
  } catch (e) {
    var result = 'b';
  }
  setTimeout(function () {
    console.log(result);
  });
}
```

Notice that `var result` is defined twice. No warning is reported.

If you change `try..catch` to `if..else` problem goes away and following warning is correctly reported:

```
'result' is already defined on line 5
```

**Comment by lukeapage** [comment]
I'm looking into this as part of my refactor

**Comment by lukeapage** [comment]
@jugglinmike 

> JSHint creates a new function scope for catch clauses, but the spec calls for a new Lexical Environment. To the best of my understanding, JSHint has no concept of Lexical Environment as distinct from function scope, so resolving this bug will be non-trivial.

Do you know what the creation of a new lexical environment affects? is it just so that the catch variable is essentially block scope? are there any other implications? I've been testing browsers and I can't see any other implications for jshint at the moment..

**Comment by lukeapage** [comment]
Fixed by #2468 - catch is now treated like any other block, not as if it is a function. Will be released in the release after 2.8.0.
